python/ray/experimental/channel.py

```diff
@@ -0,0 +1,142 @@
+import io
+import logging
+from typing import Any, Optional
+
+import ray
+from ray.util.annotations import PublicAPI
+
+# Logger for this module. It should be configured at the entry point
+# into the program using Ray. Ray provides a default configuration at
+# entry/init points.
+logger = logging.getLogger(__name__)
+
+
+def _create_channel_ref(
+    buffer_size: int,
+) -> "ray.ObjectRef":
+    """
+    Create a channel that can be read and written by co-located Ray processes.
+
+    The channel has no buffer, so the writer will block until reader(s) have
+    read the previous value. Only the channel creator may write to the channel.
+
+    Args:
+        buffer_size: The number of bytes to allocate for the object data and
+            metadata. Writes to the channel must produce serialized data and
+            metadata less than or equal to this value.
+    Returns:
+        Channel: A wrapper around ray.ObjectRef.
+    """
+    worker = ray._private.worker.global_worker
+    worker.check_connected()
+
+    value = b"0" * buffer_size
+
+    try:
+        object_ref = worker.put_object(
+            value, owner_address=None, _is_experimental_mutable_object=True
+        )
+    except ray.exceptions.ObjectStoreFullError:
+        logger.info(
+            "Put failed since the value was either too large or the "
+            "store was full of pinned objects."
+        )
+        raise
+    return object_ref
+
+
+@PublicAPI(stability="alpha")
+class Channel:
+    """
+    A wrapper type for ray.ObjectRef. Currently supports ray.get but not
+    ray.wait.
+    """
+
+    def __init__(self, buffer_size: Optional[int] = None):
+        """
+        Create a channel that can be read and written by co-located Ray processes.
+
+        Only the caller may write to the channel. The channel has no buffer,
+        so the writer will block until reader(s) have read the previous value.
+
+        Args:
+            buffer_size: The number of bytes to allocate for the object data and
+                metadata. Writes to the channel must produce serialized data and
+                metadata less than or equal to this value.
+        Returns:
+            Channel: A wrapper around ray.ObjectRef.
+        """
+        if buffer_size is None:
+            self._base_ref = None
+        else:
+            self._base_ref = _create_channel_ref(buffer_size)
```

Comment:
Do we support writing to refs from processes besides the one that created it now?

Reply:
No, I'll add a unit test showing this.